runtime/src/snapshot_utils.rs

```diff
@@ -417,43 +574,56 @@ pub fn archive_snapshot_package(
     Ok(())
 }
 
-pub fn get_snapshot_paths<P: AsRef<Path>>(snapshot_path: P) -> Vec<SlotSnapshotPaths>
+/// Get a list of SlotSnapshotPath for the given snapshots directory
+pub fn get_slot_snapshot_paths<P>(snapshots_dir: P) -> Vec<SlotSnapshotPath>
 where
-    P: fmt::Debug,
+    P: AsRef<Path>,
 {
-    match fs::read_dir(&snapshot_path) {
-        Ok(paths) => {
-            let mut names = paths
-                .filter_map(|entry| {
-                    entry.ok().and_then(|e| {
-                        e.path()
-                            .file_name()
-                            .and_then(|n| n.to_str().map(|s| s.parse::<u64>().ok()))
-                            .unwrap_or(None)
-                    })
-                })
-                .map(|slot| {
-                    let snapshot_path = snapshot_path.as_ref().join(slot.to_string());
-                    SlotSnapshotPaths {
-                        slot,
-                        snapshot_file_path: snapshot_path.join(get_snapshot_file_name(slot)),
-                    }
+    match fs::read_dir(&snapshots_dir) {
+        Ok(paths) => paths
+            .filter_map(|entry| {
+                entry.ok().and_then(|e| {
+                    e.path()
+                        .file_name()
+                        .and_then(|n| n.to_str().map(|s| s.parse::<Slot>().ok()))
+                        .unwrap_or(None)
                 })
-                .collect::<Vec<SlotSnapshotPaths>>();
-
-            names.sort();
-            names
-        }
+            })
+            .map(|slot| {
+                let snapshot_file_name = get_snapshot_file_name(slot);
+                // Yep, you read that right, the same snapshot file name is joined twice in a row.
+                // "Why" you ask?  This I do not know.
```

Comment:
i think this is just overlook from original implementation... so could you word like this?

```
// The redundant snapshot_file_name in path is unintentional and should be simplified
// kept just for compatibility.
```

the motto here is that no Qs into our own code... xD

Reply:
Done!